src/main/java/com/aws/greengrass/util/platforms/windows/WindowsRunasProcess.java

```diff
@@ -0,0 +1,450 @@
+    }
+
+    private void closeHandles() {
+        if (outPipeWriteHandle != null) {
+            try {
+                Kernel32Util.closeHandleRefs(outPipeWriteHandle, outPipeReadHandle, errPipeWriteHandle,
+                        errPipeReadHandle, inPipeReadHandle, inPipeWriteHandle);
+            } catch (Win32Exception e) {
+                // Nothing we can do. We made best effort to close resources
+            }
+        }
+        if (procInfo != null) {
+            try {
+                Kernel32Util.closeHandles(procInfo.hProcess, procInfo.hThread);
+            } catch (Win32Exception e) {
+                // Nothing we can do. We made best effort to close resources
+            }
+        }
+    }
+
+    @Override
+    public OutputStream getOutputStream() {
+        return stdin;
+    }
+
+    @Override
+    public InputStream getInputStream() {
+        return stdout;
+    }
+
+    @Override
+    public InputStream getErrorStream() {
+        return stderr;
+    }
+
+    @Override
+    public synchronized int waitFor() throws InterruptedException {
+        if (procInfo == null) {
+            throw new RuntimeException("process was not created");
+        }
+        if (exited) {
+            return exitCode;
+        }
+        if (WinBase.WAIT_FAILED == Kernel32.INSTANCE.WaitForSingleObject(procInfo.hProcess, Kernel32.INFINITE)) {
+            throw lastErrorRuntimeException();
+        }
+        return exitValue();
+    }
+
+    @Override
+    public synchronized int exitValue() {
+        if (procInfo == null) {
+            throw new RuntimeException("process was not created");
+        }
+        if (exited) {
+            return exitCode;
+        }
+        IntByReference exitCodeRef = new IntByReference();
+        if (!Kernel32.INSTANCE.GetExitCodeProcess(procInfo.hProcess, exitCodeRef)) {
+            throw lastErrorRuntimeException();
+        }
+        if (WinBase.STILL_ACTIVE == exitCodeRef.getValue()) {
+            throw new IllegalThreadStateException("process hasn't exited");
+        }
+        exited = true;
+        exitCode = exitCodeRef.getValue();
+        return exitCode;
+    }
+
+    @Override
+    public void destroy() {
+        destroyForcibly();
+    }
+
+    @Override
+    public synchronized Process destroyForcibly() {
+        if (procInfo == null || exited) {
+            return this;
+        }
+        Kernel32.INSTANCE.TerminateProcess(procInfo.hProcess, EXIT_CODE_TERMINATED);
+        closeHandles();
+        exited = true;
+        exitCode = EXIT_CODE_TERMINATED;
+        return this;
+    }
+
+    private static LastErrorException lastErrorRuntimeException() {
+        return new LastErrorException(Kernel32.INSTANCE.GetLastError());
+    }
+
+    private static ProcessCreationException lastErrorProcessCreationException(String context) {
+        return new ProcessCreationException(String.format("[%s] %s", context, Kernel32Util.getLastErrorMessage()));
+    }
```

Comment:
not sure why we have two types of exceptions that are essentially the same (GetLastError vs getLastErrorMessage)

Reply:
So I'm using LastErrorException (which is a RuntimeException defined by JNA) in the Java Process interface methods (waitfor, exitValue) because the interface doesn't declare checked exceptions.